classes/author/DAO.inc.php

```diff
@@ -0,0 +1,167 @@
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class DAO
+ * @ingroup author
+ *
+ * @see \PKP\author\Author
+ *
+ * @brief Operations for retrieving and modifying Author objects.
+ */
+
+namespace PKP\author;
+
+use APP\author\Author;
+
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\LazyCollection;
+use PKP\core\EntityDAO;
+use PKP\services\PKPSchemaService;
+use stdClass;
+
+class DAO extends EntityDAO
+{
+    /** @copydoc EntityDAO::$schema */
+    public $schema = PKPSchemaService::SCHEMA_AUTHOR;
+
+    /** @copydoc EntityDAO::$table */
+    public $table = 'authors';
+
+    /** @copydoc EntityDAO::$settingsTable */
+    public $settingsTable = 'author_settings';
+
+    /** @copydoc EntityDAO::$primarykeyColumn */
+    public $primaryKeyColumn = 'author_id';
+
+    /** @copydoc EntityDAO::$primaryTableColumns */
+    public $primaryTableColumns = [
+        'id' => 'author_id',
+        'email' => 'email',
+        'includeInBrowse' => 'include_in_browse',
+        'publicationId' => 'publication_id',
+        'seq' => 'seq',
+        'userGroupId' => 'user_group_id',
+    ];
+
+    /**
+     * Constructor
+     */
+    public function __construct(
+        PKPSchemaService $schemaService
+    ) {
+        parent::__construct($schemaService);
+    }
+
+    /**
+     * Instantiate a new DataObject
+     */
+    public function newDataObject(): Author
+    {
+        return App::make(Author::class);
+    }
+
+    /**
+     * @copydoc EntityDAO::get()
+     */
+    public function get(int $id): ?Author
+    {
+        return parent::get($id);
+    }
+
+    /**
+     * Get the total count of rows matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->count();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('a.' . $this->primaryKeyColumn)
+            ->pluck('a.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Get a collection of publications matching the configured query
+     */
+    public function getMany(Collector $query): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->select(['a.*'])
+            ->get();
+
+        return LazyCollection::make(function () use ($rows) {
+            foreach ($rows as $row) {
+                yield $this->fromRow($row);
+            }
+        });
+    }
+
+    /**
+     * @copydoc EntityDAO::fromRow()
+     */
+    public function fromRow(stdClass $row): Author
+    {
+        $author = parent::fromRow($row);
+
+        $locale = DB::table($this->table . ' as a')
```

Comment:
This means an additional query for every author fetched; it should be possible to add the missing locale column in the Collector directly.

Reply:
Still needs resolving!

Reply:
I removed it all together - I think this collector's statement `$q->select(['*', 's.locale AS submission_locale']);` is sufficient.